File: 4th semester/Systems for design and implementation/lab11WebApp/src/app/wizard/shared/wizard.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { map, Observable } from 'rxjs';
import { Wizard } from './wizard.model';

@Injectable({
  providedIn: 'root'
})
export class WizardService {
  private url = "http://localhost:8080/api/wizards"

  constructor(private http: HttpClient) { }

  getWizards(): Observable<Wizard[]> {
    return this.http.get<Array<Wizard>>(this.url).pipe(map(res => {return res['wizards']}));
  }

  getWizard(id: number): Observable<Wizard> {
    return this.getWizards().pipe(map(wizards => wizards.find(wizard => wizard.id === id)))
  }

  saveWizard(wizard: Wizard): Observable<Wizard> {
    console.log("save wizard", wizard)

    return this.http.post<Wizard>(this.url, wizard)
  }

  updateWizard(wizard: Wizard): Observable<Wizard> {
    return this.http.put<Wizard>(this.url + "/" + wizard.id, wizard)
  }

  deleteWizard(id: number): Observable<any> {
    return this.http.delete(this.url + "/" + id)
  }

  filterWizards(name: string) {
    return this.http.get<Array<Wizard>>(this.url + "/filter/" + name).pipe(map(res => {return res['wizards']}));
  }

  sortWizards() {
    return this.http.get<Array<Wizard>>(this.url + "/sort").pipe(map(res => {return res['wizards']}));
  }
}

File: 4th semester/Systems for design and implementation/lab10/SocketsProject/common/src/main/java/Service/WizardService.java
package Service;

import java.util.List;
import java.util.concurrent.Future;

import Model.ValidatorException;
import Model.Wizard;

public interface WizardService {
    String ADD = "addWizard";
    String DELETE = "deleteWizard";
    String UPDATE = "updateWizard";
    String GET = "getWizards";

    Future<Wizard> addWizard(Long id, String name, Integer age, String pet) throws ValidatorException;
    Future<Wizard> deleteWizard(Long id);
    Future<Wizard> updateWizard(Long id, String name, Integer age, String pet);
    Future<List<Wizard>> getWizards();
}
